Codecademy/gamut
# migrate layout components

### Overview


Adds the essential Storybook layout components (not navigation components) and theme switching component. Also fixes default homepage + docs ordering


Note: currently this isn't rendering properly in the preview env, but it should once the [ConnectedForm fix](https://github.com/codecademy-engineering/mono/pull/8329) is merged into its base branch 

### PR Checklist

- [x] Related to JIRA ticket: [GM-839, [GM-844](https://skillsoftdev.atlassian.net/browse/GM-844)]

#### Testing Instructions

* see the new Gamut theme working
* head to a component 
* click on the theme switcher tool
* see it working

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |




[GM-844]: https://skillsoftdev.atlassian.net/browse/GM-844?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
